docs/sources/configure-client/trace-span-profiles/dotnet-span-profiles.md

```diff
@@ -0,0 +1,72 @@
+---
+title: Span profiles with Traces to profiles for .NET
+menuTitle: Span profiles with Traces to profiles (.NET)
+description: Learn about and configure Span profiles with Traces to profiles in Grafana for .NET applications.
+weight: 103
+---
+
+# Span profiles with Traces to profiles for .NET
+
+![span-profiles screenshot](https://grafana.com/static/img/docs/tempo/profiles/tempo-profiles-Span-link-profile-data-source.png)
+
+## Before you begin
+
+Your applications must be instrumented for profiling and tracing before you can use span profiles.
+
+* Profiling: Your application must be instrumented with Pyroscope's .NET instrumentation library. If you haven't done this yet, please refer to the [.NET]({{< relref "../language-sdks/dotnet" >}}) guide.
+* Tracing: Your application must be instrumented with OpenTelemetry traces. If you haven't done this yet, please refer to the [OpenTelemetry](https://opentelemetry.io/docs/net/getting-started/) guide.
+
+{{< admonition type="note" >}}
+Currently only [OpenTelemetry manual instrumentation](https://opentelemetry.io/docs/languages/net/instrumentation/) is supported for span profiles in .NET.
+This limitation is there because Pyroscope's .NET profiler and OpenTelemetry's auto instrumentation are based on separate .NET CLR profilers.
+{{< /admonition >}}
+
+## OpenTelemetry support
+
+Pyroscope can integrate with distributed tracing systems supporting the [**OpenTelemetry**](https://opentelemetry.io/docs/instrumentation/net/getting-started/) standard which allows you to
+link traces with the profiling data, and find resource usage for specific lines of code for your trace spans
+
+{{< admonition type="note" >}}
+* Only CPU profiling is supported at the moment.
+* Because of how sampling profilers work, spans shorter than the sample interval may not be captured.
+{{< /admonition >}}
+
+
+## Configure the `Pyroscope.Tracing.OpenTelemetry` package
+
+To start collecting Span Profiles for your .NET application, you need to include [Pyroscope.Tracing.OpenTelemetry](https://github.com/grafana/pyroscope-dotnet/tree/main/Pyroscope/Pyroscope.Tracing/Pyroscope.Tracing.OpenTelemetry) in your code.
+
+This package provides a [`SpanProcessor`](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry/BaseProcessor.cs) implementation, which connects the two telemetry signals (traces and profiles) together.
+
+```shell
+dotnet add package Pyroscope.Tracing.OpenTelemetry
+```
+
+Next, we need to create and register the `PyroscopeSpanProcessor`:
+```csharp
+builder.Services.AddOpenTelemetry()
+    .WithTracing(b =>
+    {
+        b
+        .AddAspNetCoreInstrumentation()
+        .AddConsoleExporter()
+        .AddOtlpExporter()
+        .AddProcessor(new PyroscopeSpanProcessor.Builder()
+            .WithRootSpanOnly(true)
+            .Build());
+    });
```

Comment:
What do you think about removing the `WithRootSpanOnly` option? It should be `true` by default, and, honestly, I'm not very sure if we do want to make it configurable. We left this in other integrations to avoid breaking changes

Reply:
It is true by default, I added it here so that it is more visible. I don't have a strong opinion about it, I will remove the configuration option for now and add it back in the future if needed.